hepcrawl/utils.py

```diff
@@ -40,30 +44,48 @@ def unzip_xml_files(filename, target_folder):
     return xml_files
 
 
-def ftp_connection_info(ftp_host, netrc_file):
+def ftp_connection_info(ftp_host, netrc_file, passive_mode=False):
     """Return ftp connection info from netrc and optional host address."""
     if not ftp_host:
         ftp_host = netrc(netrc_file).hosts.keys()[0]
     logininfo = netrc(netrc_file).authenticators(ftp_host)
     connection_params = {
         "ftp_user": logininfo[0],
         "ftp_password": logininfo[2],
+        "ftp_passive": passive_mode,
     }
     return ftp_host, connection_params
 
 
-def ftp_list_files(server_folder, target_folder, server, user, password):
+def ftp_list_files(server_folder, target_folder, server, user, password, passive_mode=False):
     """List files from given FTP's server folder to target folder."""
-    with ftputil.FTPHost(server, user, password) as host:
+    session_factory = ftputil.session.session_factory(
+        base_class=ftplib.FTP,
+        port=21,
+        use_passive_mode=passive_mode,
+        encrypt_data_channel=True)
+
+    with ftputil.FTPHost(server, user, password, session_factory=session_factory) as host:
         files = host.listdir(host.curdir + '/' + server_folder)
-        missing_files = []
-        all_files = []
-        for filename in files:
-            destination_file = os.path.join(target_folder, filename)
-            source_file = os.path.join(server_folder, filename)
-            if not os.path.exists(destination_file):
-                missing_files.append(source_file)
-            all_files.append(source_file)
+        return list_files(server_folder, target_folder, files)
+
+
+def local_list_files(local_folder, target_folder):
+    """List files from given package folder to target folder."""
+    files = os.listdir(local_folder)
+    return list_files(local_folder, target_folder, files)
+
+
+def list_files(remote_folder, target_folder, files):
```

Comment:
Maybe this should be called `list_missing_files` instead?
The `files` parameter would be better called `file_names`.
It feels to me that this is doing too many things at the same time... can you think of a way to split getting the list of missing files, with adding the paths to the files? (so there's no need to return both `all_files` and `missing_files`, making it easier to understand, test and use)